src/amo/components/AddonCompatibilityError/index.js

```diff
@@ -0,0 +1,94 @@
+/* eslint-disable react/no-danger */
+import { oneLine } from 'common-tags';
+import React, { PropTypes } from 'react';
+import { compose } from 'redux';
+import { connect } from 'react-redux';
+
+import {
+  INCOMPATIBLE_FIREFOX_FOR_IOS,
+  INCOMPATIBLE_NOT_FIREFOX,
+  INCOMPATIBLE_OVER_MAX_VERSION,
+  INCOMPATIBLE_UNDER_MIN_VERSION,
+} from 'core/constants';
+import translate from 'core/i18n/translate';
+import { sanitizeHTML } from 'core/utils';
+
+import './style.scss';
+
+
+export class AddonCompatibilityErrorBase extends React.Component {
+  static propTypes = {
+    i18n: PropTypes.object.isRequired,
+    lang: PropTypes.string.isRequired,
+    maxVersion: PropTypes.string.isRequired,
+    minVersion: PropTypes.string.isRequired,
+    reason: PropTypes.string.isRequired,
+    userAgentInfo: PropTypes.object.isRequired,
+  }
+
+  static defaultProps = {
+    userAgentInfo: {},
+  }
+
+  render() {
+    const {
+      i18n,
+      lang,
+      maxVersion,
+      minVersion,
+      reason,
+      userAgentInfo,
+    } = this.props;
+    const downloadUrl = `https://www.mozilla.org/${lang}/firefox/`;
+    let message;
+
+    if (typeof maxVersion === 'undefined') {
+      throw new Error('maxVersion is required; it cannot be undefined');
+    }
+    if (typeof minVersion === 'undefined') {
+      throw new Error('minVersion is required; it cannot be undefined');
+    }
```

Comment:
I think this would be a better place to check if `reason` is undefined because that's definitely a programming mistake.